### PR TITLE
Store repository data in data path if not previously set

### LINKED ISSUE #12292: snap packge pre-seeds /var/snap/gitea/$version/ in install page breaking snap updates
- Gitea version (or commit ref): v1.12.2 (stable snap channel)
- Git version:
- Operating system: Ubuntu Core 18
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

When starting the /install page for a fresh installation the writable snap directories are properly set as default (assuming from the $SNAP_DATA variable which contains these versions) i.e.:

    /var/snap/gitea/4837/data/gitea.db (when using sqlite)
    /root/snap/gitea/4837/gitea-repositories
    /var/snap/gitea/4837/lfs
    /var/snap/gitea/4837/log

The problem here is that the version numbers end up in app.ini. 
Upgrading the snap will fail because when upgrading the version number of the directory name is bumped and read access to the old versioned directory gets denied (the versioned dirs are fully bound to the binary versions of a snap)

Snapd creates symlinks by default to avoid such issues, the pre-seeded values in the install page should instead point to the directory names involving these symlinks i.e.:

    /var/snap/gitea/current/data/gitea.db
    /root/snap/gitea/current/gitea-repositories
    /var/snap/gitea/current/lfs
    /var/snap/gitea/current/log
    ...

...this way upgrades and roll-back will be possible.
...

### PR DESCRIPTION
**Pull request by techknowlogick**
Fixes #12292 - stores snap repo data in common directory so it isn't set to store in previous snap version path.

Previously it is set to store repo data in $HOME if not set, instead this sets it to store in `AppDataPath` if not set. Checked for breaking on arch & alpine and doesn't break them, however as this changes a default setting I've marked it as breaking.

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
it's nice cleanup for modules/setting/repository.go too ...

**Review by silverwind — CHANGES_REQUESTED** [review]
Please update cheat sheet and app.example.ini (adding same comment there)

**Comment by techknowlogick** [comment]
@silverwind thanks for reminder. I have now update docs.

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -42,8 +42,8 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
 
 ## Repository (`repository`)
 
-- `ROOT`: **~/gitea-repositories/**: Root path for storing all repository data. It must be
-   an absolute path.
+- `ROOT`: **data/gitea-repositories/**: Root path for storing all repository data. It must be
+   an absolute path. By default it is stored in a sub-directory of `APP_DATA_PATH`
```
Comment (silverwind) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
   an absolute path. By default it is stored in a sub-directory of `APP_DATA_PATH`.
```

**Comment by silverwind** [comment]
Still missing from `app.example.ini`, just copy the text from the cheat sheet.

**Comment by techknowlogick** [comment]
@silverwind ah yes, I've also added in one unrelated part of docs too as it was empty and next to `ROOT` so it was easy enough (although I could include it in separate PR if needed)

**Review by silverwind — APPROVED** [review]

**Pull request merged**